CentralHub.Api/Controllers/MeasurementController.cs

```diff
@@ -0,0 +1,77 @@
+using CentralHub.Api.Model;
+using CentralHub.Api.Model.Requests.Localization;
+using CentralHub.Api.Model.Responses.AggregatedMeasurements;
+using CentralHub.Api.Services;
+using Microsoft.AspNetCore.Mvc;
+
+namespace CentralHub.Api.Controllers;
+
+[ApiController]
+[Route("/measurements")]
+public sealed class MeasurementController : ControllerBase
+{
+    private readonly ILogger<MeasurementController> _logger;
+
+    private readonly IAggregatedMeasurementRepository _aggregatorRepository;
+
+    public MeasurementController(
+        ILogger<MeasurementController> logger,
+        IAggregatedMeasurementRepository aggregatorRepository)
+    {
+        _logger = logger;
+        _aggregatorRepository = aggregatorRepository;
+    }
+
+    [HttpPost("add")]
+    public async Task AddMeasurements(AddMeasurementsRequest addMeasurementsRequest, CancellationToken token)
+    {
+        await _aggregatorRepository.AddMeasurementsAsync(
+            addMeasurementsRequest.TrackerId,
+            addMeasurementsRequest.Measurements,
+            token);
+    }
+
+    [HttpGet("get")]
+    public async Task<GetAggregatedMeasurementsResponse> GetMeasurements(int roomId, CancellationToken token)
+    {
+        var aggregatedMeasurements = await _aggregatorRepository.GetMeasurementsAsync(roomId, token);
+
+        if (aggregatedMeasurements == null)
+        {
+            return GetAggregatedMeasurementsResponse.CreateUnsuccessful();
+        }
+
+        return GetAggregatedMeasurementsResponse.CreateSuccessful(
+            aggregatedMeasurements.Select(am => new AggregatedMeasurements(
+            am.MeasurementDtoId,
+            am.StartMeasurementTime,
+            am.EndMeasurementTime,
+            am.MeasurementCount,
+            am.BluetoothMedian,
+            am.BluetoothMean,
+            am.BluetoothMax,
+            am.BluetoothMin,
+            am.BluetoothCount,
+            am.WifiMedian,
+            am.WifiMean,
+            am.WifiMax,
+            am.WifiMin,
+            am.WifiCount,
+            am.RoomDtoId)));
+    }
```

Comment:
Not sure why we create a new object with the old object instead of using the same object.
Am i missing something here?

Reply:
The aggregatedMeasurements we get at first consists of Dto objects,  we need to convert this before we provide it in the response.